client/src/components/Related.jsx

```diff
@@ -0,0 +1,374 @@
+                        : null
+                    }
+                  >{`$${item.default_price}`}</p>
+                  <p style={{ display: 'inline' }}>{style[index].salePrice}</p>
+                  <span className="star">&#9733;&#9733;&#9733;&#9733;&#9733;</span>
+                </section>
+              )
+            })}
+          </TinySlider>
+          <div className='controls'>
+            <button
+              onClick={arrowClick}
+              id='first-btn'
+              type='button'
+              style={arrowDiff === 0 ? { display: 'none' } : null}
+            >
+              ❮
+            </button>
+            <button
+              onClick={arrowClick}
+              id='second-btn'
+              type='button'
+              style={
+                arrowDiff === style.length - 3 ? { display: 'none' } : null
+              }
+            >
+              ❯
+            </button>
+          </div>
+        </div>
+
+        <div
+          className='modal-container'
+          style={show ? null : { display: 'none' }}
+        >
+          <button onClick={closeModal}>×</button>
+          <table>
+            <thead>
+              <tr>
+                <th>{currentProduct.name}</th>
+                <th className='centerText'>Characteristic</th>
+                <th className='left-tick'>{compareProduct.name}</th>
+              </tr>
+            </thead>
+            {currentProduct.features.map((item, index) => {
+              {
+                console.log('current modal window information', item)
+              }
+              if (item) {
+                return (
+                  <thead key={index}>
+                    <tr>
+                      <th>{currentProduct.features.includes(item) && '✔'}</th>
+                      <th className='centerText'>{item}</th>
+                      <th className='left-tick'>
+                        {compareProduct.features.includes(item) && '✔'}
+                      </th>
+                    </tr>
+                  </thead>
+                )
+              }
+            })}
+            {compareProduct.features.map((item, index) => {
+              console.log(
+                'modal window information',
+                item,
+                currentProduct.features.includes(item)
+              )
+              if (item && !currentProduct.features.includes(item)) {
+                return (
+                  <thead key={index}>
+                    <tr key={index}>
+                      <th>{currentProduct.features.includes(item) && '✔'}</th>
+                      <th className='centerText'>{item}</th>
+                      <th className='left-tick'>
+                        {compareProduct.features.includes(item) && '✔'}
+                      </th>
+                    </tr>
+                  </thead>
+                )
+              }
+            })}
+          </table>
+        </div>
+      </>
+    )
+  }
+}
+
+export default Related
```

Comment:
Is there a way to break down this Related components into different subcomponents?
- getting all the information with one subcomponent
-  rendering information on another subcomponent?

Reply:
If you guys think it is necessary I can break it into two components.

Reply:
I don't think it is ABSOLUTELY necessary. If it's functional, it's functional! MVP